src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -982,6 +983,83 @@ class TcgStatics {
     }
   }
 
+  static holon_pokemon_energy(Object delegate, Integer rainbowEnergyCount) {
+    delegate.globalAbility { Card thisCard->
+      delayed {
+          before PLAY_CARD, {
+            def validPokemon = thisCard.player.pbg.all.findAll {it.cards.energyCount()}
+            def holonAlreadyPlayed = bg.em().retrieveObject("Holon_Pokemon_Energy") == bg.turnCount
```

Comment:
These checks also shouldn't be needed if using PlayEnergy. `allowAttach` should work fine for checking valid Pokémon in my testing.

Reply:
Thanks for the suggestion, I've used `allowedAttach` and it appears to be working great